lib/membrane_live/event_service.ex

```diff
@@ -0,0 +1,208 @@
+  Depending on a response ends the event or sets new timer with notifiaction.
+  Should be called when user sends response to EventService notification `last_viewer_answer`.
+  """
+  @spec send_response(user_response(), event_id()) :: {:response, event_id(), user_response()}
+  def send_response(user_response, event_id) do
+    send(EventService, {:response, event_id, user_response})
+  end
+
+  @doc """
+  Ends the event and marked is as finished in database.
+  """
+  @spec send_kill(event_id()) :: {:kill, event_id()}
+  def send_kill(event_id), do: send(EventService, {:kill, event_id})
+
+  @impl true
+  def handle_call({:start_room, event_id}, _from, state) do
+    case :global.whereis_name(event_id) do
+      :undefined ->
+        {:ok, pid} = Room.start(event_id, name: {:global, event_id})
+        state = put_in(state, [:pid_to_id, pid], event_id)
+        Process.monitor(pid)
+        {:reply, pid, state}
+
+      pid ->
+        {:reply, {:already_started, pid}, state}
+    end
+  end
+
+  @impl true
+  def handle_info({:event, action, event_id}, state) do
+    previous_users_number = get_in(state.events, [event_id, :users_number]) || 0
+    timer_ref = get_in(state.events, [event_id, :timer_ref])
+
+    users_number =
+      case action do
+        :join -> previous_users_number + 1
+        :leave -> previous_users_number - 1
+      end
+
+    timer_ref =
+      case users_number do
+        0 ->
+          cancel_action(timer_ref)
+          kill_action(event_id)
+
+        1 ->
+          cancel_action(timer_ref)
+          notify_action(event_id)
+
+        2 ->
+          cancel_action(timer_ref)
+          nil
+
+        _other ->
+          nil
+      end
+
+    event = %{users_number: users_number, timer_ref: timer_ref}
+    {:noreply, put_in(state, [:events, event_id], event)}
+  end
+
+  @impl true
+  def handle_info({:notify, event_id}, state) do
+    MembraneLiveWeb.Endpoint.broadcast!("event:" <> event_id, "last_viewer_active", %{
+      timeout: @response_timeout
+    })
+
+    timer_ref = kill_action(event_id)
+    {:noreply, put_in(state, [:events, event_id, :timer_ref], timer_ref)}
+  end
+
+  @impl true
+  def handle_info({:response, event_id, :leave}, state) do
+    __MODULE__.send_kill(event_id)
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_info({:response, event_id, :stay}, state) do
+    users_number = get_in(state.events, [event_id, :users_number])
+    timer_ref = get_in(state.events, [event_id, :timer_ref])
+
+    cancel_action(timer_ref)
+
+    timer_ref =
+      case users_number do
+        0 -> kill_action(event_id)
+        1 -> notify_action(event_id)
+        _other -> nil
+      end
+
+    {:noreply, put_in(state, [:events, event_id, :timer_ref], timer_ref)}
+  end
+
+  @impl true
+  def handle_info({:kill, event_id}, state) do
+    pid = :global.whereis_name(event_id)
+
+    {event, state} = pop_in(state, [:events, event_id])
+    {_event_id, state} = pop_in(state, [:pid_to_id, pid])
```

Comment:
I think we could use [`delete_in` here](https://hexdocs.pm/bunch/Bunch.Access.html#delete_in/2).
```suggestion
    state = delete_in(state, [:pid_to_id, pid])
```

Reply:
I would have to use `Bunch.Access.delete_in/2` and I don't think it's worth it. What do you think?

Reply:
It was only a suggestion. I think `delete_in` is more suitable here, but no problem if you don't want to use it.